testing/cuda/scan_by_key.cu

```diff
@@ -98,6 +98,24 @@ void TestScanByKeyDevice(ExecutionPolicy exec)
     ASSERT_EQUAL(cudaSuccess, err);
   }
   ASSERT_EQUAL(d_output, h_output);
+
+  // in-place scans: keys/values aliasing
+  thrust::inclusive_scan_by_key(h_keys.begin(), h_keys.end(), h_vals.begin(), h_output.begin());
+  inclusive_scan_by_key_kernel<<<1,1>>>(exec, d_keys.begin(), d_keys.end(), d_vals.begin(), d_keys.begin());
+  {
+    cudaError_t const err = cudaDeviceSynchronize();
+    ASSERT_EQUAL(cudaSuccess, err);
+  }
+  ASSERT_EQUAL(d_keys, h_output);
```

Comment:
Nitpick, no change requested: I would prefer if we put the whole test in its own block rather than only the check, as that helps readability a lot
```suggestion
  { // in-place scans: keys/values aliasing
    thrust::inclusive_scan_by_key(h_keys.begin(), h_keys.end(), h_vals.begin(), h_output.begin());
    inclusive_scan_by_key_kernel<<<1,1>>>(exec, d_keys.begin(), d_keys.end(), d_vals.begin(), d_keys.begin());
    cudaError_t const err = cudaDeviceSynchronize();
    ASSERT_EQUAL(cudaSuccess, err);
    ASSERT_EQUAL(d_keys, h_output);
  }
```